### PR TITLE
Missing property fix

### PR DESCRIPTION
**Pull request by aozgaa**
Implements #13071.

Note that there is considerably more work to do for stubbing out missing methods if we want to do any type inference on the parameters. There are additionally issues with inferring type parameters for the stubbed method from its type arguments.

Thoughts @mhegazy @billti @DanielRosenwasser ?

### CODE REVIEW

File: src/services/codefixes/fixAddMissingMember.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+            return undefined;
+        }
+
+        let typeString = "any";
+
+        if (token.parent.parent.kind === SyntaxKind.BinaryExpression) {
+            const binaryExpression = token.parent.parent as BinaryExpression;
+
+            const checker = context.program.getTypeChecker();
+            const widenedType = checker.getBaseTypeOfLiteralType(checker.getTypeAtLocation(binaryExpression.right));
```
Comment (mhegazy) [inline src/services/codefixes/fixAddMissingMember.ts]: you also want to widen it. i.e. `null` => `any`

File: src/services/codefixes/fixAddMissingMember.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+            changes: [{
+                fileName: sourceFile.fileName,
+                textChanges: [{
+                    span: { start: startPos, length: 0 },
+                    newText: `${token.getFullText(sourceFile)}: ${typeString};`
+                }]
+            }]
+        },
+        {
+            description: formatStringFromArgs(getLocaleSpecificMessage(Diagnostics.Add_index_accessor_for_missing_property_0), [token.getText()]),
```
Comment (DanielRosenwasser) [inline src/services/codefixes/fixAddMissingMember.ts]: 1. "index accessor" -> "index signature"
2. You need a test for this fix.

Comment (aozgaa) [inline src/services/codefixes/fixAddMissingMember.ts]: changed and test added

**Comment by mhegazy** [comment]
@aozgaa after merging in master please port this to release-2.2

**Pull request merged**